ubuntu-advantage

```diff
@@ -14,10 +14,25 @@ CA_CERTIFICATES=${CA_CERTIFICATES:="/usr/sbin/update-ca-certificates"}
 SNAPD=${SNAPD:="/usr/lib/snapd/snapd"}
 KERNEL_VERSION=$(uname -r)
 
+check_result() {
+    if [ "$1" -ne "0" ]; then
+        echo "ERROR"
+        if [ -n "$2" ]; then
+            echo "$2" >&2
+        fi
+        exit $1
+    else
+        echo "OK"
+    fi
+}
+
```

Comment:
You could actually put all the logic in this function:

```bash
check_result() {
    local result output
    result=0
    output=$("$@" 2>&1) || result=$?
    if [ $result -ne 0 ]; then
        echo "ERROR"
        if [ "$output" ]; then
            echo "$output"
        fi
    else
        echo "OK"
    fi   
}
```

Then the call site would be just ```check_call apt-get install -y snapd``` (and you can drop `result` and `output` there)